openwisp_controller/config/api/urls.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.urls import path
+
+from . import views as api_views
+
+app_name = 'openwisp_controller'
+
+
+def get_api_urls(api_views):
+    """
+    returns:: all the API urls of the config app
+    """
+    if getattr(settings, 'OPENWISP_CONTROLLER_API', True):
+        return [
+            path(
+                'controller/template/',
+                api_views.template_list,
+                name='api_template_list',
+            ),
+            path(
+                'controller/template/<str:pk>/',
+                api_views.template_detail,
+                name='api_template_detail',
+            ),
+            path(
+                'controller/template/<str:pk>/configuration/',
+                api_views.download_template_config,
+                name='api_download_template_config',
+            ),
+            path('controller/vpn/', api_views.vpn_list, name='api_vpn_list',),
+            path(
+                'controller/vpn/<str:pk>/', api_views.vpn_detail, name='api_vpn_detail',
```

Comment:
SN | controller/vpn/str:pk/ | Status
-- | -- | --
1 | [SU] Can view any vpn | Pass
2 | [O1A] Can view only org1 vpns (not even shared org) | Pass
3 | [O1O] Cannot view any vpn | Pass
4 | [SU][O1A] Correct data of the vpn shown | Pass
5 | [SU][O1A][O1O] Can only delete templates if they have correct privilege | Pass